nautobot/docs/development/core/ui-best-practices.md

```diff
@@ -0,0 +1,190 @@
+    <div class="flex-grow px-8 py-10">
+        Content
+    </div>
+</div>
+```
+
+The above component is generally well-built, with Bootstrap helpers and utilities used for proper styling. With some slight differences this could be an actual go-to implementation for a hypothetical component. But in this specific case we can do better. Let's see the last example.
+
+```html
+<div class="card">
+    <div class="card-body">
+        Content
+    </div>
+</div>
+```
+
+This is the exact same component as the previous ones, but instead of building styles up from the bottom it re-uses basic Bootstrap Card component, and require no further styling. Real world scenarios may not be as simple as this, but it servers as a good example of the idea that's being laid out here.
+
+## Style
+
+### Style overrides
+
+As mentioned, it is recommended to use off-the-shelf components and elements, but sometimes they may require various style adjustments to meet specific requirements, be it a custom color, size, spacing, etc.
+
+Bootstrap offers a broad range of re-usable styles, from single-style [Helpers](https://getbootstrap.com/docs/5.3/helpers/) and [Utilities](https://getbootstrap.com/docs/5.3/utilities/) to ready-to-use layouts and components. On top of that, Nautobot also provides its own extensions documented in [New Nautobot Custom UI APIs](../apps/migration/from-v2/new-nautobot-custom-ui-apis.md#extended-bootstrap-utilities).
+
+When styling elements, we recommend starting with base high-level (component) classes and, if needed, override their specific styles with Bootstrap or Nautobot helpers and utilities.
+
+To expand on the previous Card component example, let's say we need to render a card with green background and border, i.e. a "success card":
+
+```html
+<div class="card bg-success-subtle border-success">
+    <div class="card-body">
+        Success
+    </div>
+</div>
+```
+
+### Custom styles
+
+In case available helpers and utilities are not sufficient to style a component, there are two possible ways to approach this problem. To find the more suitable out of them depends on a particular use case. Let's answer these questions first:
+
+1. Is the style used in a single or - at most - very few places?
+2. Is the style simple and does not require any combined selectors?
+3. If the style involves a custom defined color - should the color be the same for application's light and dark theme?
+
+If the answer to all these questions is yes, then we advise using HTML `style` attribute for simplicity as well as coupling the style tightly with a specific element.
+
+Otherwise, define a custom CSS code within `<style>` tag in document head, preferably inside a `{% block extra_styles %}` Django template block. If developing for the core and custom styles are indented to be used throughout multiple pages, consider adding them to the main `nautobot.scss` file.
```

Comment:
```suggestion
Otherwise, define a custom CSS code within `<style>` tag in document head, preferably inside a `{% block extra_styles %}` Django template block. If developing for the core and custom styles are intended to be used throughout multiple pages, consider adding them to the main `nautobot.scss` file.
```